File: GeoJSON4EntityFramework/Base/GeoJsonElement.vb
Public MustInherit Class GeoJsonElement
    <JsonProperty(PropertyName:="type", Order:=1)>
    Public ReadOnly Property TypeName As String
        Get
            Return [GetType].Name
        End Get
    End Property

    Public Function Serialize(Optional prettyPrint As Boolean = False) As String
        Return GeoJsonSerializer.Serialize(Me, prettyPrint)
    End Function

    Public Function Serialize(settings As JsonSerializerSettings) As String
        Return GeoJsonSerializer.Serialize(Me, settings)
    End Function
End Class

File: GeoJSON4EntityFrameworkCS/Base/GeoJsonElement.cs
using Newtonsoft.Json;

namespace GeoJSON4EntityFramework.Base
{
    public class GeoJsonElement<T>
    {
        [JsonProperty(PropertyName = "type", Order = 1)]
        public string TypeName
        {
            get { return typeof (T).Name; }
        }
    }
}
